### ISSUE TITLE
[Bug]: Some vehicles spawn with a flat white rectangle

### ISSUE THREAD

**Issue by ghost** [Yimura/YimMenu#3418]
### Describe the bug

There are some vehicles that spawn with a flat white rectangle, that look like a paper sheet, and when driving it follows us, like if it was part of the vehicle.

I notice this with the vehicle called " Hakuchou Drag ".

### Steps To Reproduce

1. Start your game
2. Inject YimMenu
3. Go to online or story mode
4. Spawn the vehicle " Hakuchou Drag " (for example)
5. Notice that there is a big paper sheet following you

### Expected Behavior

Expected behaviour would be that there is no flat white rectangle appearing.

### Logs

No log to provide.

### Screenshots / Video

![Paper Sheet](https://github.com/user-attachments/assets/a5b3027b-f2e7-4abc-b81b-c69e197da5da)

### Store

Rockstar

### Language

English

### YimMenu Language

English (US)

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### Coload

- [X] I've tested this without co-loading anything.

**Comment by ghost** [comment]
Also forgot to mention that :
- The prop does not collide with anything
- It disappears after driving like 25 meters

**Comment by Yimura** [comment]
Are you spawning the vehicle with max upgrades?

**Comment by ghost** [comment]
Yes I do, and I just tried without maxing it and the white prop is not here.
However, when spawning it unmaxed and then manually maxing it, it appears back.
So it's seems like it's a vehicle mod that's broken or something.

**Comment by Yimura** [comment]
Yeah this is kind of Rockstar's fault and requires us to manually check if the vehicle has a mod on it that's exclusive to next-gen consoles. If these next-gen mods are applied then a white rectangle will appear.

**Comment by Yimura** [comment]
We are supposedly checking for it correctly before maxing the vehicle however it appears that these invalid mods are still being applied:
https://github.com/YimMenu/YimMenu/blob/c15c57335f6026a554eac087a8285b6ab952e553/src/util/vehicle.cpp#L582

**Comment by ghost** [comment]
No idea if that can be a thing but maybe they made a mistake when coding the native ?

**Comment by TwystedMind** [comment]
The floating rectangle appears specifically if the vehicle has any HSW upgrades applied (since they are not available on PC).  If you remove the HSW modifications under the LSC tab while in a vehicle, it will remove the rectangle.

**Comment by lonelybud** [comment]
are you using an old saved vehicle file?

**Comment by ghost** [comment]
Nope, that's a brand new I spawned already maxed.

**Comment by gir489returns** [comment]
Related to #3448?

**Comment by ghost** [comment]
I don't think it's related to that issue

**Comment by ghost** [comment]
Yep also just had that bug. Grabbed a car off the street. Maxed it with the menu and it had a white rectangle
![Uploading Bild_2024-07-28_161120415.png…]()
